sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryUtils.java

```diff
@@ -403,7 +403,7 @@ private static List<TableFieldSchema> toTableFieldSchema(Schema schema) {
       }
       if (type.getTypeName().isCollectionType()) {
         type = Preconditions.checkArgumentNotNull(type.getCollectionElementType());
-        if (type.getTypeName().isCollectionType() || type.getTypeName().isMapType()) {
+        if (type.getTypeName().isCollectionType() && !type.getTypeName().isMapType()) {
```

Comment:
BQ supports arrays of maps, but not arrays of other collections?

Reply:
BigQuery only supports arrays of other supported types, excepts any collection/map type. 

This change enables storing a simple map or an array of maps (after flattening them) as a repeated struct field. 

A separated change may help to support flattening other collection types (arrays of arrays for example).

Reply:
removed this change.